### DISCUSSION TITLE
Newbie Event confusion

### DISCUSSION THREAD

**Discussion by dahlb** [jason0x43/hacs-hubitat#113] [Q&A] [answered]
First off thanks for a great integration.

I am new to HA and am trying to port my rule machine over because my hubitat has started running slow.  I found the dev-tools full event created by pushing my light switch,

{
    "event_type": "hubitat_event",
    "data": {
        "device_id": "485",
        "device_name": "Dog light switch",
        "attribute": "held",
        "value": "2",
        "description": null,
        "type": null,
        "hub": "dc8c14da"
    },
    "origin": "LOCAL",
    "time_fired": "2021-04-05T18:30:34.995928+00:00",
    "context": {
        "id": "4d35f5429ed89314f4b60777fb0dabff",
        "parent_id": null,
        "user_id": null
    }
}

but can't figure out how to convert that hubitat device_id into the HA device_id, or why the below works in an automation for a trigger but looks so different from the event I was able to see produced.

device_id: 225062d72ed2c7685913f770e15d73ad
domain: hubitat
platform: device
type: held
subtype: '2'

Any insight or referral would be greatly appreciated, my current goal is to make a blueprint for use with my light switches as having one automation per button configured seem likely to grow unmanagable (work in progress https://github.com/dahlb/home-assistant/blob/main/blueprints/inovelli-switch.yaml) but I'm stuck on the condition without a HA device_id

**Comment by jason0x43** [comment]
HA has a couple different ways of handling hardware events. Changes that involve a state change, like a switch changing from off to on, generate entity state changes in HA. In an automation or in Node Red, you’d respond to the state change. Changes that are ephemeral, like a button being pressed, generate events. You would respond to the event in an automation or Node Red.(Internally both actions generate events, but you don’t typically deal with a state change event directly.)

**Reply by dahlb** [reply]
I understand some things generate state change and others having no state only generate events, my situation is only about events.  Sorry, I didn't make my question more clear initially.  I have a hubitat_event from your integration and it has a hubitat device_id but I haven't seen any way to convert that device_id to a HA device_id for convenience in automation it would be great.  Have you considered adding the HA device_id to the hubitat_event data? or am I missing a HA built in function that could provide the conversion?  I didn't see the hubitat device_id as an attribute on the entities, so I'm at a loss as to how to convert between the two.

**Comment by dahlb** [comment, marked as answer]
got the blueprint working using the raw hubitat device_id for now, https://github.com/dahlb/home-assistant/blob/main/blueprints/inovelli-switch.yaml. I'll put in a feature request for the hubitat_event to include the HA device_id in addition to the hubitat device_id because then the blue print could use a selector
